src/fetch_business_partners_info.py

```diff
@@ -0,0 +1,104 @@
+
+
+def make_consulta_socio_url(name):
+    return 'http://www.consultasocio.com/q/sa/%s' % normalize_name(name).replace(' ', '-')
+
+
+def extract_info_from_page(url, info, driver, error_msg):
+    # Open the page using Selenium in order to collect the full HTML to be parsed by BeautifulSoup
+    driver.get(url)
+    driver.switch_to.default_content()
+    soup = BeautifulSoup(driver.page_source, 'html.parser')
+
+    # Somethimes google can ask for captcha, so you should type it!
+    if 'To continue, please type the characters below:' in soup.text:
+        while 'To continue, please type the characters below:' in soup.text:
+            driver.switch_to.default_content()
+            soup = BeautifulSoup(driver.page_source, 'html.parser')
+            time.sleep(1)
+
+    # Collect the information if there is no error
+    if error_msg.lower() not in soup.title.text.lower():
+        for partner in soup.findAll('span', attrs={'class': 'socio'}):
+            info['partners'].append(partner.text)
+        for company_info in soup.findAll('p', attrs={'class': 'number'}):
+            info['business'].append(company_info.find('span', {'class': 'text'}).text)
+        try:
+            info['cache_info'] = soup.find('div', attrs={'id': 'google-cache-hdr'}).text
+        except:
+            info['cache_info'] = str(datetime.datetime.now())
+        print(info)
+        return info
+
+
+def fetch_business_infos():
+    file_base_name = '{}-congressperson-business-partners.json'.format(datetime.date.today().strftime('%Y-%m-%d'))
+    output_file_path = os.path.join(os.path.abspath('../data'), file_base_name)
+
+    # Load cached data if file exists
+    result = {}
+    if os.path.exists(output_file_path):
+        result = json.loads(open(output_file_path, 'r').read())
+
+    # Collect the information from all congress person
+    civil_names_df = pd.read_csv('../data/2016-12-07-congressperson-civil-names.xz')
+
+    driver = webdriver.Chrome(selenium_driver_path())
+
+    original_site_blocked = False
+    id = 0
+    collected_count = 0
+    total_count = len(civil_names_df['civil_name'].unique())
+    for name in civil_names_df['civil_name'].unique():
+        name = normalize_name(name).upper()
+
+        print('(%d/%d) %s' % (id, total_count, name), sep=' ', end='\n')
+        if name in result:
+            collected_count += 1
+        else:
+            default_info = {'partners': [], 'business': []}
+            if not original_site_blocked:
+                info = extract_info_from_page(make_consulta_socio_url(name), default_info, driver,
+                                              'Uso — ConsultaSocio.com')
+                original_site_blocked = info is None
+            if info is None:
+                info = extract_info_from_page(make_google_cache_url(name), default_info, driver, 'error')
+            if info:
+                result[name] = info
+                open(output_file_path, 'w').write(json.dumps(result))
```

Comment:
> Is possible to create CSV using 1 to N items?

There are two strategies. One is exemplified in `YYYY-MM-DD-companies.xz` and the other one in `YYYY-MM-DD-congressperson_relatives.xz`. 

If you have a lot of different columns (eg `YYYY-MM-DD-companies.xz`) you can create a very large amount of columns, even if most of them are blank. In the companies dataset each company has around 20 columns with regular data (name, tarde name, address, etc.), plus 200 other columns for `secondary_activity` and `secondary_activity_code` (ie `secondary_activity_1` and `secondary_activity_1_code`, `secondary_activity_2` and `secondary_activity_2_code`, `secondary_activity_3` and `secondary_activity_3_code` up to 100).

If you don't have too much basic info, let's say, congressperson name, congressperson document, and just want to add a list of relative for each one of them, you just repeat lines (eg `YYYY-MM-DD-congressperson_relatives.xz`):

```csv
congressperson_document, congressperson_name, relative
42, Fulano de Tal, Ciclano
42, Fulano de Tal, Beltrano
42, Fulano de Tal, Etc
```

I think this is the best option for you.

Yet you can fit data in two columns: one dataset with the descriptive data (eg. congressperson name and document, an unique ID), and a second dataset with this unique ID (repeat lines as much as needed) and the extra info — both datasets are linked in a foreign key strategy. Works, but adds extra steps to analyze it later.